optuna/storages/_rdb/models.py

```diff
@@ -87,6 +86,50 @@ def all(cls, session: orm.Session) -> List["StudyModel"]:
         return session.query(cls).all()
 
 
+class StudyDirectionModel(BaseModel):
+    __tablename__ = "study_directions"
+    __table_args__: Any = (UniqueConstraint("study_id", "objective"),)
+    study_direction_id = Column(Integer, primary_key=True)
+    direction = Column(Enum(StudyDirection), nullable=False)
+    study_id = Column(Integer, ForeignKey("studies.study_id"))
+    objective = Column(Integer, default=0)
+
+    study = orm.relationship(
+        StudyModel, backref=orm.backref("directions", cascade="all, delete-orphan")
+    )
+
+    @classmethod
+    def find_by_study_and_objective(
```

Comment:
I'm wondering whether we need this method. It's not used (only tested). Did you intend to use it in the near future? I otherwise think we can simplify the API by removing it. Objectives are usually few and `where_study` might do as a generalized version. As a matter of fact, it seems like you're using `where_study` already in the `RDBStorage` followed by indexing `[0]`.

Reply:
Thank you for pointing that out. Surely this API is not necessary. Initially, I added this API as I thought that directions could be given in `Dict[int, StudyDirection]` instead of just sequences and could partially call `set_study_direction` forgiven axes. However, for backward compatibility, we only pass a sequence to `set_study_direction`, so we don't need this API now. I'll delete it.

Reply:
Uh, I've noticed that it is necessary for the MO integration. Please check it in #1994 later. I'll revert the deletion.

Reply:
I committed https://github.com/optuna/optuna/pull/1994/commits/6a564a6a0cdbe71f7cd62f438f0dd35e99676905 and `StudyDirectionModel.find_by_study_and_objective` is used in optuna/storages/_rdb/storage.py:L243-245.

Reply:
Sorry for making you go back and forth. Okay, let's leave it as is. Actually, you could write the query out, especially if the method is only used once, but the way we define methods directly in the models aren't clearly communicates so it's not a strong opinion. Let me resolve this one.